infra/gcp/ensure-prod-storage.sh

```diff
@@ -180,6 +180,13 @@ empower_service_account_to_artifacts \
     $(svc_acct_email "${PROMOTER_TEST_PROD_PROJECT}" "${PROMOTER_SVCACCT}") \
     "${PROMOTER_TEST_STAGING_PROJECT}"
 
+# Special case: grant the image promoter test service account OWNER access to
+# their testing project (used for running e2e tests for the promoter auditing
+# mechanism).
+empower_service_account_as_owner \
```

Comment:
I'm not super happy about owner here.  What actual permissions does it need?  Is there no smaller scope we can apply?

Reply:
I don't know the exact scopes but I do know the gcloud commands that the service account must execute as part of the test (manually scraped from `~/go/src/sigs.k8s.io/k8s-container-image-promoter/test-e2e/cip-auditor/cip-auditor-e2e.go`). There are many gcloud invocations that the service account must run as part of the test (not in order):

```
gcloud logging logs list
gcloud logging read
gcloud logging logs delete
gcloud beta error-reporting events delete
gcloud run services list
gcloud run services delete
gcloud run services add-iam-policy-binding ...
gcloud run services describe
gcloud run deploy
gcloud pubsub subscriptions list
gcloud pubsub subscriptions delete
gcloud pubsub subscriptions seek
gcloud pubsub subscriptions create
gcloud pubsub topics list
gcloud pubsub topics delete
gcloud pubsub topics create
gcloud projects add-iam-policy-binding ...
```

Of the above commands, the one that ran first was `gcloud projects add-iam-policy-binding k8s-gcr-audit-test-prod --member=serviceAccount:service-375340694213@gcp-sa-pubsub.iam.gserviceaccount.com --role=roles/iam.serviceAccountTokenCreator` which failed with

```
ERROR: (gcloud.projects.add-iam-policy-binding) User [k8s-infra-gcr-promoter@k8s-gcr-audit-test-prod.iam.gserviceaccount.com] does not have permission to access project [k8s-gcr-audit-test-prod:getIamPolicy] (or it may not exist): Cloud Resource Manager API has not been used in project 375340694213 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudresourcemanager.googleapis.com/overview?project=375340694213 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
```

Reply:
I'll give YOU owner on GCR_AUDIT_TEST_PROD_PROJECT and you figure the smallest (within reason) set of roles for the audit test?

Hopefully it's 4-5, based on that list - like pubsub admin, run admin, etc.

When we go live we can scope it down even more for the prod project

Reply:
@thockin I've managed to reduce it down to 7 admin-ish permissions. The auditor e2e test is now passing here https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/172 as well. PTAL

Reply:
update the PR comment and title?